sui/docs/link-token.md

```diff
@@ -0,0 +1,533 @@
+# Link Token Documentation
+
+## Overview
+
+This document explains how to link custom tokens on Sui to other chains using the Interchain Token Service (ITS).
+
+For detailed ITS design specifications and architecture, see **[ARC-1: ITS Hub Multi-Chain Token Linking](https://github.com/axelarnetwork/arcs/blob/main/ARCs/ARC-1.md)**. <!-- skip-check -->
+
+The token linking feature enables:
+
+- Linking custom tokens deployed on Sui with tokens on trusted chains
+- Supporting tokens with different decimal precisions through automatic scaling
+
+## How It Works
+
+The token linking process involves two key message types that ITS Hub uses to coordinate the linking:
+
+### Token Metadata Registration
+
+Before linking tokens, ITS Hub needs to know about each token's details (address and decimals) on both chains.
+
+### Token Linking
+
+Once metadata is registered, you can link tokens by specifying which tokens on different chains should be connected.
+
+## Link Token Flow
+
+```
+1. User has access to Token A (Source Chain) and Token B (Destination Chain)
+2. User → ITS Source Chain: registerCoinMetadata(Token A)
+3. User → ITS Destination Chain: registerTokenMetadata(Token B)
+4. User → ITS Source Chain: registerCustomCoin() → Deploys Token Manager A on Source Chain
```

Comment:
On the sui side do we not also need to call either `register_coin_from_info` or `register_coin_from_metadata` as well?

Reply:
Yes, if Sui is the remote chain. Or, registered as a custom coin (`register_custom_coin`) if it's the source chain. However, this part is just the basic GMP overview. I will try to make that more clear.